posthog/src/test/java/com/posthog/android/PostHogTest.java

```diff
@@ -397,6 +400,37 @@ protected boolean matchesSafely(GroupPayload payload) {
             );
   }
 
+  @Test
+  public void createsSessionIdIfNotSet() {
+    assertThat(properties.sessionId()).isNullOrEmpty();
+    assertThat(persistence.sessionLastTimestamp()).isNull();
+    posthog.capture("test");
+    assertThat(persistence.sessionLastTimestamp()).isInstanceOf(Instant.class);
+    assertThat(properties.sessionId()).isInstanceOf(String.class);
+  }
+
+  @Test
+  public void generatesNewSessionIdIfExpired() {
```

Comment:
Should we add one more test that's the same as this but with a custom sessionExpiration time?

Reply:
It's probably worth having a test that verified `properties.sessionId()` is assigned the value of the session id